ts/utils/fetch.ts

```diff
@@ -114,3 +129,246 @@ export const constantPollingFetch = (
 
   return retriableFetch(retryWithTransient404s, shouldAbort)(timeoutFetch);
 };
+
+/**
+ * Decorates the current fetch with LolliPOP headers and http-signature
+ */
+export function lollipopFetch(
+  timeout: Millisecond = fetchTimeout,
+  maxRetries: number = fetchMaxRetries,
+  lollipopConfig: LollipopConfig,
+  keyInfo: KeyInfo
+) {
```

Comment:
I would try to treat this function like `AbortableFetch` and `timeoutFetch` then pass a `fetch` as a parameter to the function instead of the parameters to construct it.
Something like that:
```suggestion
export lollipopFetch = (
  lollipopConfig: LollipopConfig,
  keyInfo: KeyInfo
) =>(fetchWithTimeout: fetch)=>{
```